main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala

```diff
@@ -89,6 +89,8 @@ object KindedAst {
 
     case class ApplyDef(symUse: Ast.DefSymUse, exps: List[Expr], itvar: Type, tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
 
+    case class ApplyLocalDef(sym: Symbol.VarSym, exps: List[Expr], itvar: Type.Var, tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
```

Comment:
I think magnus wanted `itvar` but every `tvar` and `evar` are also instantiated so it feels very arbitrary. can we do `arrowTvar`